### PR TITLE
Add Jira syncing to this repo

### PR DESCRIPTION
**Pull request by emmyoop**
Add jira tracking within this repo by calling workflows here.

Need to transition off the `dbt-labs/jira-actions` repo onto here in all other repos.

This will be useful for tracking with the contractor creating issues in this repo.

### CODE REVIEW

**Comment by ChenyuLInx** [comment]
Always great to see things get removed 🥳

**Comment by leahwicz** [comment]
Let's talk about this before we merge. Originally, Kyle moved these Actions here from the jira-actions repo with the intention that they would live here and we would archive the jira-actions repo. Some of our repos point to the Actions in jira-actions but some also point here. We need to consolidate to the jira-actions repo before we merge this if that is what we want to do.

@emmyoop do you think we should keep the Actions here or in the other repo? I don't have a strong feeling either way

**Comment by emmyoop** [comment]
@leahwicz  my impression was it had gone the other way.  I learn towards having it separate, honestly.  

However I also recall a previous team meeting where we made a decision to leave everything in actions unless there are very good reason to have the workflow/action in its own repo.  This use case does not seem to have a "good reason".

With that said, I'll update this PR to revert the jira actions and create new workflows to call them within this action repo so it still gets hooked up to jira.  

I'll then open issues in any repos calling the `jira-actions` workflows to update to point here so this gets tracked and not lost.  This is extra important since I know @callum-mcdata is using these workflows now as well so it's not restricted to our team.

**Review by leahwicz — APPROVED** [review]

**Pull request merged**
